### DISCUSSION TITLE
In what case would Response.statusCode be null?

### DISCUSSION THREAD

**Discussion by abstraq** [cfug/dio#1479] [Q&A]
It is my understanding that if you have a [Response](https://pub.dev/documentation/dio/latest/dio/Response-class.html) object, the request reached the server and the server responded. Since the server will always respond with a status code I am confused in what case would the statusCode property can be null.

Thanks!

**Comment by AlexV525** [comment]
Response is mutable, so let's say you have an interceptor that might return an empty response with only specific messages, and that would be a certian case for no status code.
